packages/flutter/lib/src/rendering/object.dart

```diff
@@ -669,6 +672,10 @@ class PaintingContext extends ClipContext {
   /// [RenderObject.alwaysNeedsCompositing] property to return true. That informs
   /// ancestor render objects that this render object will include a composited
   /// layer, which, for example, causes them to use composited clips.
+  @Deprecated(
+    'Use pushOpacityLayer instead to avoid forced compositing. '
+    'This feature was deprecated after v2.13.0-0.0.pre.'
+  )
```

Comment:
What about adding a {needsCompositing = false} parameter to this method and it only will add a layer if it needs compositing?

Reply:
I don't see any uses in google3 or in flutter/tests, which technically means we could change it with just a breaking change announcement.

Reply:
we could call it `pushAlpha` instead, which accurately describes the value it takes...

Reply:
Alpha is fine, but it has lower resolution than opacity. Eventually these things are stored as floats in both Skia and Impeller because GPUs tend to take float values for their shaders.

Reply:
sorry, I'm just talking about the name. The Paint/saveLayer API can only accept an alpha

Reply:
...currently...

We'll be supporting higher precision color info in the engine (SkPaint/Impeller Paint both already do so, with the Impeller object only taking float components - DlPaint currently takes int colors, but I will be enhancing it to allow float colors as well soon).

Reply:
time for Color2

Reply:
We may actually be able to swap out the backing implementing and add new APIs to handle this.